src/device-state/preload.ts

```diff
@@ -114,5 +115,23 @@ export async function loadTargetFromFile(
 				error: e,
 			});
 		}
+	} finally {
+		const targetPath = appsJsonBackup(appsPath);
+		if (!(await fsUtils.exists(targetPath))) {
```

Comment:
I acknowledge that this current code could still exhibit the issue it tries to mitigate if another apps.json file appears since it won't create a backup if a backup already exists. 

I can't find anything in our docs about apps.json but I believe this file is only created when you preload a device so it would be really weird if someone preloaded a device again after it has ran making this issue not that big of a deal. Is this correct?

Reply:
Is not really a problem, as you cannot preload a device, you preload an image which then you flash into a device, so modification of apps.json should be a one time operation.

Reply:
great! no further thought needed then. Thanks for clarifying.